### PR TITLE
[MRG] Fixing path and environment variables issues + tests on windows

### PR DESCRIPTION
**Pull request by aabadie**
This PR partly fix the issue #809 

But it maybe worth to merge. Summary of changes:
* fixed Environment variable split character for windows as it uses ';'
* replaced '/' with os.sep when required
* fixed permission issues when removing files in test_atlas.py/test_utils.py
* fixed file:// prefix using urllib crossplatform dedicated function "pathname2url"

### CODE REVIEW

File: nilearn/datasets/tests/test_func.py [hunk 1]
```diff
@@ -407,7 +408,7 @@ def test_fetch_mixed_gambles():
         mgambles = func.fetch_mixed_gambles(n_subjects=n_subjects,
                                             data_dir=tmpdir, url=local_url,
                                             verbose=0, return_raw_data=True)
-        datasetdir = os.path.join(tmpdir, "jimura_poldrack_2012_zmaps/")
+        datasetdir = os.path.join(tmpdir, "jimura_poldrack_2012_zmaps" + os.sep)
```
Comment (AlexandreAbraham) [inline nilearn/datasets/tests/test_func.py]: Here you can just remove os.sep I think.

File: nilearn/datasets/func.py [hunk 2]
```diff
@@ -1138,7 +1138,7 @@ def fetch_abide_pcp(data_dir=None, n_subjects=None, pipeline='cpac',
 
     # Go into specific data folder and url
     data_dir = os.path.join(data_dir, pipeline, strategy)
-    url = '/'.join([url, 'Outputs', pipeline, strategy])
+    url = os.sep.join([url, 'Outputs', pipeline, strategy])
```
Comment (AlexandreAbraham) [inline nilearn/datasets/func.py]: This is very surprising, I've never seen a URL with '\' to separate elements, even on Windows.

File: nilearn/datasets/utils.py [hunk 3]
```diff
@@ -16,6 +16,11 @@
 import zipfile
 from .._utils.compat import _basestring, cPickle, _urllib, md5_hash
 
+# Environment variables splitter is different on Windows
+if sys.platform == 'win32':
+    ENV_SPLITTER = ';'
+else:
+    ENV_SPLITTER = ':'
```
Comment (AlexandreAbraham) [inline nilearn/datasets/utils.py]: Good catch!

File: nilearn/datasets/func.py (hunk 2 above)
Comment (aabadie) [inline nilearn/datasets/func.py]: indeed, I overlooked this part. Maybe my change is breaking the default behavior. Will try again.

File: nilearn/datasets/utils.py (hunk 3 above)
Comment (lesteve) [inline nilearn/datasets/utils.py]: [os.pathsep](https://docs.python.org/2/library/os.html#os.pathsep) can probably be used

File: nilearn/datasets/tests/test_utils.py [hunk 4]
```diff
@@ -99,13 +99,14 @@ def test_get_dataset_dir():
...
     assert os.path.exists(data_dir)
+    os.chmod(no_write, 0o600)
     shutil.rmtree(data_dir)
 
     # Verify exception for a path which exists and is a file
     test_file = os.path.join(tmpdir, 'some_file')
     with open(test_file, 'w') as out:
         out.write('abcfeg')
-    assert_raises_regex(OSError, 'Not a directory',
+    assert_raises_regex(OSError, 'Not a directory|introuvable',
```
Comment (lesteve) [inline nilearn/datasets/tests/test_utils.py]: :sob: I don't think we want to go down the road of supporting all locale. Maybe you can catch the exception and look at exc.errno?

File: nilearn/datasets/tests/test_atlas.py [hunk 5]
```diff
@@ -126,13 +126,15 @@ def test_get_dataset_dir():
     # Non writeable dir is returned because dataset may be in there.
     assert_equal(data_dir, no_write)
     assert os.path.exists(data_dir)
+    os.chmod(no_write, 0o600) # Enable write permissions back
```
Comment (lesteve) [inline nilearn/datasets/tests/test_atlas.py]: PEP8 (two spaces before inline comment). Also mention why you reenabling write permissions (to be able to delete it I guess)

File: nilearn/datasets/utils.py (hunk 3 above)
Comment (aabadie) [inline nilearn/datasets/utils.py]: > os.pathsep can probably be used

Thanks for the hint, I'll change this.

File: nilearn/datasets/tests/test_utils.py (hunk 4 above)
Comment (aabadie) [inline nilearn/datasets/tests/test_utils.py]: > I don't think we want to go down the road of supporting all locale. Maybe you can catch the exception and look at exc.errno?

Good idea, will do this.

**Comment by aabadie** [comment]
All comments should be adressed in 2ef8080

**Comment by bthirion** [comment]
LGTM

**Comment by GaelVaroquaux** [comment]
LGTM. Merging. Thank you!

**Pull request merged**